core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -125,11 +125,21 @@ public void appendTimeoutInsight(InsightBuilder insight) {
   @CheckReturnValue
   final Runnable setStream(ClientStream stream) {
     ClientStreamListener savedListener;
+    ClientStream oldStream = null;
+    boolean cancelOldStream = false;
+
     synchronized (this) {
       // If realStream != null, then either setStream() or cancel() has been called.
       if (realStream != null) {
+        oldStream = realStream;
+        cancelOldStream = listener != null;
+      }
+      if (oldStream != null && !cancelOldStream) {
         return null;
       }
+      if (cancelOldStream) {
+        oldStream.cancel(Status.CANCELLED.withDescription("Replaced by a new Stream"));
```

Comment:
No. We've already chosen a stream. We can't change to a different stream now. I don't understand why this is being done this way, especially since #1537 said that the stream passed into this function would be the one cancelled.